lib/models/builder.js

```diff
@@ -163,17 +162,29 @@ module.exports = CoreObject.extend({
    * @method trapSignals
    */
   trapSignals: function() {
-    if (!signalsTrapped) {
```

Comment:
i believe we may want to guard `trapSignals` from being called unbalanced from `_cleanupSignals`. Likely throwing if done so.